### PR TITLE
Fixes #23195 - Audit associations on creation and destruction (#5455)

### PR DESCRIPTION
**Pull request by ares**
Also includes a bit of refactoring to the AuditAssociations module.

### CODE REVIEW

**Comment by ares** [comment]
PR has been reviewed in https://github.com/theforeman/foreman/pull/5455

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 1]
```diff
@@ -461,6 +461,7 @@ def save_vm(uuid, attr)
       if attr.has_key?(:volumes_attributes)
         vm.volumes.each do |vm_volume|
           volume_attrs = attr[:volumes_attributes].values.detect {|vol| vol[:id] == vm_volume.id}
+      binding.pry
```
[the remark this answers is not in the document]
Comment (mmoll) [reply app/models/compute_resources/foreman/model/vmware.rb]: @ares :point_up:

[the remark this answers is not in the document]
Comment (ares) [reply app/models/compute_resources/foreman/model/vmware.rb]: oh, I cherry-picked on unclean repo and didn't notice, sorry, will redo on clean checkout to be sure

**Comment by ares** [comment]
re-cherry-picked

**Comment by mmoll** [comment]
merged, díky @ares!

**Pull request merged**